subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/TaskFilePropertyCompareStrategy.java

```diff
@@ -33,32 +33,30 @@
 
 public enum TaskFilePropertyCompareStrategy {
     ORDERED(new OrderSensitiveTaskFilePropertyCompareStrategy()),
-    UNORDERED(new OrderInsensitiveTaskFilePropertyCompareStrategy(true)),
-    OUTPUT(new OrderInsensitiveTaskFilePropertyCompareStrategy(false));
```

Comment:
Yes, we need to distinguish between them. But not on the `TaskFilePropertySpec` level. We only need to distinguish on the `FileCollectionSnapshotBuilder` level, which is set by the different snapshotters (generic and runtime/compile classpath) and does not need to be passed via `TaskFilePropertySpec`.